common/src/main/java/io/crate/types/TypeCompatibility.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.types;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class TypeCompatibility {
+
+    private static final TypeCompatibility INCOMPATIBLE = new TypeCompatibility(null, false);
+
+    @Nullable
+    public static DataType<?> getCommonType(DataType<?> firstType, DataType<?> secondType) {
+        TypeCompatibility compatibility = compatibility(firstType, secondType);
+        if (!compatibility.isCompatible()) {
+            return null;
+        }
+        return compatibility.getCommonType();
+    }
+
+    private static TypeCompatibility compatible(DataType<?> commonSuperType, boolean coercible) {
+        return new TypeCompatibility(commonSuperType, coercible);
+    }
+
+    private static TypeCompatibility compatibility(DataType<?> fromType, DataType<?> toType) {
+        if (fromType.equals(toType)) {
+            return compatible(toType, true);
+        }
+
+        if (fromType.equals(UndefinedType.INSTANCE)) {
+            return compatible(toType, true);
+        }
+
+        if (toType.equals(UndefinedType.INSTANCE)) {
+            return compatible(fromType, false);
+        }
+
+        String fromTypeBaseName = fromType.getTypeSignature().getBaseTypeName();
+        String toTypeBaseName = toType.getTypeSignature().getBaseTypeName();
+        if (fromTypeBaseName.equals(toTypeBaseName)) {
+            // If given types share the same base, e.g. arrays, parameter types must be compatible.
+            if (!fromType.getTypeParameters().isEmpty() || !toType.getTypeParameters().isEmpty()) {
+                return typeCompatibilityForParametrizedType(fromType, toType);
+            }
+            return compatible(fromType, false);
```

Comment:
Shouldn't this be compatible?

Reply:
It's compatible as both base names are equal and no parameters e.g. `text` and `text`.